COCOAPODS.md

```diff
@@ -1,150 +1,381 @@
+dependencies on Pod libraries stored in the CocoaPods repository or locally as well as use a multiplatform project with 
+native targets as a CocoaPods dependency (Kotlin Pod).
 
- - Such a dependency can be included in a Podfile of an Xcode project and automatically built (and rebuilt)
- along with this project. As a result, importing to Xcode is simplified since there is no need to
- write corresponding Gradle tasks and Xcode build steps manually.
+You can manage Pod dependencies right in Intellij IDEA enjoying its benefits of working with code, such as code
+ highlighting and completion. You can build the whole Kotlin project with Gradle as a whole not switching to Xcode. 
+ Go to Xcode only when you need to write Swift/Objective-C code or run your application on a simulator or device.
  
- - When building from Xcode, you can use CocoaPods libraries without writing
- .def files manually and setting cinterop tool parameters. In this case, all required parameters can be
- obtained from the Xcode project configured by CocoaPods.
+Depending on your purpose, you can add dependencies between:
+* [A Kotlin project and a Pod library from the CocoaPods repository](#add-a-dependency-on-a-pod-library-from-the-cocoapods-repository)
+* [A Kotlin project and a Pod library stored locally](#add-a-dependency-on-a-pod-library-stored-locally)
+* [A Kotlin Pod and an Xcode project with one target](#add-a-dependency-between-a-kotlin-pod-and-xcode-project-with-one-target) 
+or [several targets](#add-a-dependency-between-a-kotlin-pod-with-an-xcode-project-with-several-targets)
+* [Several Kotlin Pods and an Xcode project](#add-dependencies-between-multiple-kotlin-pods-and-an-xcode-project)
 
-For an example of CocoaPods integration, refer to the
-[`cocoapods`](https://github.com/JetBrains/kotlin-native/tree/master/samples/cocoapods) sample.
+>You can also add dependencies between a Kotlin Pod and multiple Xcode projects. However, in this case you need to add a 
+>dependency by calling `pod install` manually for each Xcode project. In other cases, it's done automatically.
+{:.note}
 
-## CocoaPods Gradle plugin
+## Install the CocoaPods dependency manager and plugin
 
-The CocoaPods support is implemented in a separate Gradle plugin: `org.jetbrains.kotlin.native.cocoapods`.
+1. Install the [CocoaPods dependency manager](https://cocoapods.org/).
+    
+    <div class="sample" markdown="1" theme="idea" mode="ruby" data-highlight-only>
+    
+    ```ruby
+    $ sudo gem install cocoapods
+    ```
+    
+    </div>
+
+2. Install the [`cocoapods-generate`](https://github.com/square/cocoapods-generate) plugin.
+
+    <div class="sample" markdown="1" theme="idea" mode="ruby" data-highlight-only>
+    
+    ```ruby
+    $ sudo gem install cocoapods-generate
+    ```
+    
+    </div>
+    
+3. In `build.gradle.kts` (or `build.gradle`) of your IDEA project, apply the CocoaPods plugin as well as the Kotlin
```

Comment:
3. In `build.gradle.kts` (or `build.gradle`) of your IntelliJ IDEA project, apply the CocoaPods plugin as well as the Kotlin Multiplatform plugin it is based on.

" apply the CocoaPods plugin as well as the Kotlin Multiplatform plugin it is based on."
this could be clearer with what it means? -- apply how, what is the Multiplatform plugin it is based on?

Reply:
CocoaPods plugin is based on Multiplatform plugin. That's why we need to apply both. I can remove "it is based on" if it seems confusing. The idea is to explain somehow why you need to apply multiplatform plugin but it's not critical.

Reply:
Thanks! I think, as it is not described that it is based on it right now, it is superfluous information and "it is based on" is a good thing to cut as they don't need to know this and they may be like me... easily confused :)

Reply:
✅